src/lib/lib.c

```diff
@@ -79,7 +79,7 @@ int memset_s(void *dest, size_t dest_size, int data, size_t count)
 	return arch_memset_s(dest, dest_size, data, count);
 }
 
-#if !__XCC || !XCHAL_HAVE_HIFI3 || !CONFIG_LIBRARY
+#if !__XCC__ || !XCHAL_HAVE_HIFI3 || CONFIG_LIBRARY
 void *__vec_memcpy(void *dst, const void *src, size_t len)
```

Comment:
Probably easiest just to declare this weak, no?  That allows this fallback to be declared unconditionally without needing to know a-priori which targets define it.  Doesn't look like SOF has a standard token for that, though Zephyr will.  So something like:

```c
// Maybe put this in a SOF header somewhere...
#ifndef __ZEPHYR__
#define __weak __attribute__((__weak__))
#endif

__weak void *__vec_memcpy(void *dst, const void *src, size_t len)
{
    return memcpy(dst, src, len);
}
```

Reply:
I will try this out, but its dependent on xcc actually handling it properly.